discord/message.py

```diff
@@ -793,6 +794,38 @@ def flatten_handlers(cls):
     return cls
 
 
+class MessagePin:
+    """Represents information about a pinned message.
+
+    .. versionadded:: 2.7
+    """
+
+    def __init__(
+        self,
+        state: ConnectionState,
+        channel: MessageableChannel,
+        data: MessagePinPayload,
+    ):
+        self._state: ConnectionState = state
+        self._pinned_at: datetime.datetime = utils.parse_time(data["pinned_at"])
+        self._message: Message = state.create_message(
```

Comment:
Doesn't this cache the message? If it does, I feel like we should avoid it because usually people want recent messages cached, not really old pins